components/scream/src/share/io/scream_scorpio_interface.F90

```diff
@@ -1359,22 +1355,55 @@ end function get_dimlen
   !  grid_write_darray_1d: Write a variable defined on this grid
   !
   !---------------------------------------------------------------------------
-  subroutine grid_write_darray_1d(filename, varname, var_data_ptr)
+  subroutine grid_write_darray_float(filename, varname, buf, buf_size)
     use piolib_mod, only: PIO_setframe
     use piodarray,  only: PIO_write_darray
 
     ! Dummy arguments
-    character(len=*),   intent(in) :: filename       ! PIO filename
-    character(len=*),   intent(in) :: varname
-    type (c_ptr),       intent(in) :: var_data_ptr
+    character(len=*),    intent(in) :: filename       ! PIO filename
+    character(len=*),    intent(in) :: varname
+    integer(kind=c_int), intent(in) :: buf_size
+    real(kind=c_float),  intent(in) :: buf(buf_size)
 
     ! Local variables
-    real(rtype), dimension(:), pointer         :: var_data_real
-    integer(kind=c_int), dimension(:), pointer :: var_data_int
-    type(pio_atm_file_t), pointer      :: pio_atm_file
-    type(hist_var_t), pointer          :: var
-    integer                            :: ierr,var_size
-    logical                            :: found
+
+    type(pio_atm_file_t), pointer :: pio_atm_file
+    type(hist_var_t), pointer     :: var
+    integer                       :: ierr,var_size
+    logical                       :: found
+
+    call lookup_pio_atm_file(trim(filename),pio_atm_file,found)
+    call get_var(pio_atm_file,varname,var)
+
+    ! Set the timesnap we are reading
+    call PIO_setframe(pio_atm_file%pioFileDesc,var%piovar,int(max(1,pio_atm_file%numRecs),kind=pio_offset_kind))
+
+    ! We don't want the extent along the 'time' dimension
+    var_size = SIZE(var%compdof)
+
+    ! Now we know the exact size of the array, and can shape the f90 pointer
+    call pio_write_darray(pio_atm_file%pioFileDesc, var%piovar, var%iodesc, buf, ierr)
+    call errorHandle( 'eam_grid_write_darray_float: Error writing variable '//trim(varname),ierr)
+  end subroutine grid_write_darray_float
+  subroutine grid_write_darray_double(filename, varname, buf, buf_size)
+    use piolib_mod, only: PIO_setframe
+    use piodarray,  only: PIO_write_darray
+
+    ! Dummy arguments
+    character(len=*),    intent(in) :: filename       ! PIO filename
+    character(len=*),    intent(in) :: varname
+    integer(kind=c_int), intent(in) :: buf_size
+    real(kind=c_double), intent(in) :: buf(buf_size)
+
+    ! Local variables
+
+    real(kind=c_float) :: sbuf(buf_size)
+    type(pio_atm_file_t), pointer :: pio_atm_file
+    type(hist_var_t), pointer     :: var
+    integer                       :: ierr,var_size
+    logical                       :: found
+
+    print *, "write_darray_double"
```

Comment:
Leftover from debugging?

Reply:
Ah, yes. We'll remove it whenever we put up a PR next.